### ISSUE TITLE
pathSuffix matches partial segments from the wrong side

### ISSUE THREAD

**Issue by jrudolph** [spray/spray#1056]
I.e. the following test passes:

```scala
"""pathSuffix("foo" / "bar" ~ "baz")""" should {
  val test = testFor(pathSuffix("foo" / "bar" ~ "baz") { echoUnmatchedPath })
  "accept [/orders/barbaz-suffix/foo]" in test("/orders/-suffix")
}
```

The same problem exists in akka-http.

It could be that a fix would also allow to revisit the decision that `pathSuffix` consumes the unmatched path from the end. If you look at the example above (`"foo" / "bar" ~ "baz"`) it seems as if the concatenation with slash would be handled differently from the general `~`.

**Comment by sirthias** [comment]
I agree that the behavior of `pathSuffix` is not the most intuitive.
However, it behaves as "specified".
The scaladoc specifically says:

```
Note that, for efficiency reasons, the given PathMatcher must match the desired suffix in reversed-segment order, ...
```

**Comment by jrudolph** [comment]
I know and I understand the implementation but it isn't really that useful if it leaves half of a segment from the middle of the original path in the unmatched path afterwards.

**Comment by sirthias** [comment]
Ah, I see.
Yeah, that's really not that great.
Maybe we need to indeed bite the bullet and implement a "proper", even if slow, suffix match.
